File: TransCoder_tokenized_test_set_functions/java/DYNAMIC_PROGRAMMING_SET_11_EGG_DROPPING_PUZZLE.java
static int eggDrop ( int n, int k ) {
  if ( k == 1 || k == 0 ) return k;
  if ( n == 1 ) return k;
  int min = Integer . MAX_VALUE;
  int x, res;
  for ( x = 1;
  x <= k;
  x ++ ) {
    res = Math . max ( eggDrop ( n - 1, x - 1 ), eggDrop ( n, k - x ) );
    if ( res < min ) min = res;
  }
  return min + 1;
}

File: TransCoder_tokenized_test_set_functions/python/DYNAMIC_PROGRAMMING_SET_11_EGG_DROPPING_PUZZLE.py
def eggDrop ( n, k ) :
    if ( k == 1 or k == 0 ) :
        return k
    if ( n == 1 ) :
        return k
    min = sys . maxsize
    for x in range ( 1, k + 1 ) :
        res = max ( eggDrop ( n - 1, x - 1 ), eggDrop ( n, k - x ) )
        if ( res < min ) :
            min = res
    return min + 1
